app/src/main/java/nl/tudelft/cs4160/identitychain/connection/Communication.kt

```diff
@@ -0,0 +1,400 @@
+package nl.tudelft.cs4160.identitychain.connection
+
+import android.util.Log
+
+import com.google.protobuf.ByteString
+
+import java.io.UnsupportedEncodingException
+import java.security.KeyPair
+import java.util.Arrays
+import java.util.HashMap
+
+import nl.tudelft.cs4160.identitychain.Peer
+import nl.tudelft.cs4160.identitychain.block.TrustChainBlock
+import nl.tudelft.cs4160.identitychain.block.ValidationResult
+import nl.tudelft.cs4160.identitychain.connection.network.NetworkCommunication
+import nl.tudelft.cs4160.identitychain.database.TrustChainDBHelper
+import nl.tudelft.cs4160.identitychain.main.MainActivity
+import nl.tudelft.cs4160.identitychain.message.MessageProto
+
+import nl.tudelft.cs4160.identitychain.Peer.bytesToHex
+import nl.tudelft.cs4160.identitychain.block.TrustChainBlock.GENESIS_SEQ
+import nl.tudelft.cs4160.identitychain.block.TrustChainBlock.UNKNOWN_SEQ
+import nl.tudelft.cs4160.identitychain.block.TrustChainBlock.createBlock
+import nl.tudelft.cs4160.identitychain.block.TrustChainBlock.sign
+import nl.tudelft.cs4160.identitychain.block.TrustChainBlock.validate
+import nl.tudelft.cs4160.identitychain.block.ValidationResult.ValidationStatus.*
+import nl.tudelft.cs4160.identitychain.message.MessageProto.Message.newBuilder
+
+
+/**
+ * Class that is responsible for the communication.
+ */
+abstract class Communication(private val dbHelper: TrustChainDBHelper, private val keyPair: KeyPair, val listener: CommunicationListener) {
```

Comment:
did you just convert this or did you change it?

Reply:
`connectToPeer` has a `payload` parameter now, instead of looking up the default "Hello World" payload